WebKit export of https://bugs.webkit.org/show_bug.cgi?id=188189

darsh7807 opened the pull request:
WebKit export from bug: [Perform a microtask checkpoint before creating a custom element](https://bugs.webkit.org/show_bug.cgi?id=188189)

nikosandronikos committed:
Convert SVG 1.1 marker tests to reftests.

Correct result of marker tests because SVG 2 changes marker overflow auto to be visible

Update to resolve anti-aliasing differences.

Also addressed some other minor comments made by AmeliaBR.

ericwilligers committed:
SVG Text Parsing

Test that properties support the full grammar in
https://svgwg.org/svg2-draft/text.html

Blink/Firefox/Edge/Safari do not support
- shape-inside
- shape-subtract
- text-decoration-fill
- text-decoration-stroke

Safari/Edge do not support
- inline-size

Edge does not support
- shape-margin

jugglinmike committed:
[FileAPI] Remove reference to non-existent file

The file `FileAPI/url/url-with-tags.window.js` is not currently present
in the repository nor was it present when the reference was introduced
[1]. Remove it to avoid triggering spurious request failures during test
execution.

[1] 9d8a1855092e90b2974453d6181c7f985170deba

aethanyc committed:
Do not span column-span:all element across all columns if it's under different block formatting context.

multicol-span-all-004-ref.html is the same as multicol-span-all-004.html except
for the "column-span" value in h3.

Differential Revision: https://phabricator.services.mozilla.com/D12192

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1506163
gecko-commit: a5e00e5b4635a19e9fb4ed3566565504338384c6
gecko-integration-branch: mozilla-inbound
gecko-reviewers: bz

jugglinmike committed:
fixup! [FileAPI] Remove reference to non-existent file

ericwilligers committed:
[selectors] :defined CSS selector

The :defined pseudo-class must match any element that is defined.
https://html.spec.whatwg.org/multipage/semantics-other.html#selector-defined

The selector is being added to
[CSS Selectors Level 4](https://drafts.csswg.org/selectors/)
https://github.com/w3c/csswg-drafts/issues/2258#issuecomment-440756225

darsh7807 committed:
added perform-microtask-checkpoint-before-construction test

darsh7807 committed:
added perform-microtask-checkpoint-before-construction test

ericwilligers committed:
CSS: WPT for :first-child :last-child

Promote our :first-child :last-child invalidation layout test to WPT.

https://drafts.csswg.org/selectors-4/#the-first-child-pseudo

Change-Id: I3c2e1af82ef2dac58ec5bfe642b156289457bb22
Reviewed-on: https://chromium-review.googlesource.com/c/1348873
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Eric Willigers <ericwilligers@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610696}

fred-wang commented on custom-elements/perform-microtask-checkpoint-before-construction.html:
```diff
@@ -0,0 +1,144 @@
+<meta name="help" content="https://html.spec.whatwg.org/multipage/parsing.html#adoption-agency-algorithm">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/custom-elements-helpers.js"></script>
```
mmh, it seems you ignored my comment https://github.com/web-platform-tests/wpt/pull/12447#discussion_r232938117 on the other PR. I thought "/resources/custom-elements-helper.js" gets the element at the root, so I think it should be "resources/custom-elements-helper.js" instead (without the initial /). Not sure why this is still working...

cc @Ms2ger

darsh7807 replied:
@fred-wang  Just checked this  test is passing on webkit gtk

fred-wang replied:
OK, but can you please fix the links to custom-elements-helper.js here and the one added in https://github.com/web-platform-tests/wpt/pull/12447 ?

fred-wang replied:
OK, now I understand how the bad path was "fixed"... :-/

@darsh7807 Next time, please reply to questions we ask during review requests and be more explicit about how you addressed them. The move you silently did in https://github.com/web-platform-tests/wpt/pull/12447/commits/e342aed77e168325e86db3577c949c41b7f18f70 was not obvious from the commit message. Thank you.

darsh7807 replied:
Hi, sorry for the delay in reply I had build issues on webkit, I have now corrected the file path now,

fred-wang reviewed (commented):
Tests pass on Firefox but the second test fails on Chromium. Maybe @domenic can confirm this is expected and will be fixed in future Chromium release.

https://travis-ci.org/web-platform-tests/wpt/jobs/459042902
https://travis-ci.org/web-platform-tests/wpt/jobs/459042903#L625

Have you tried again with a trunk build of WebKit, just to be sure?

autofoolip committed:
Update interfaces/filter-effects.idl (#14224)

Source: https://github.com/tidoust/reffy-reports/blob/72b7f9d/whatwg/idl/filter-effects.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/459330148

ewilligers committed:
[selectors] :enabled and :disabled CSS selectors (#14200)

:enabled matches UI elements that are in an enabled state.
:disabled matches UI elements that are in a disabled state.
https://drafts.csswg.org/selectors-4/#enableddisabled

qiuzhong committed:
Remove the timeout in async_test for webrtc and xhr tests

Some async_test tests in webrtc and xhr with timeout are missed during
removing all the timeout in async_test. As part of goal #11120, this tests
should be updated too.

qiuzhong committed:
Remove timeout in async_test for geolocation-API tests

This async_test was missed during removing all the timeout from async_test.
Add `<meta name=timeout content=long>` in the html head could probably fix
this.
Related: #14118

xiuqijix committed:
Add test for "Allow delaying autoplay until video elements become visible (#12732)

chromium-wpt-export-bot committed:
Add tests that subclassing {Transform,Writable}Stream works (#14228)

Change-Id: Iba151bf31367c06c94311402a9fc550d7bb87a6d

ricea committed:
Make the ReadableStream subclassing test match the others (#14235)

ReadableStream had a simple test that it was extendable. Replace it with
a more thorough test in the style of the tests that WritableStream and
TransformStream have.

domenic commented:
@tkent-google or @hayatoito can you check that this failure is expected and will be fixable in Chrome?

eqrion committed:
Add tests for 'pointer-events: none' on the root element.

Depends on D12500

Differential Revision: https://phabricator.services.mozilla.com/D12622

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1508789
gecko-commit: 56b6151b48228e19e2e076053294b78f75880046
gecko-integration-branch: autoland
gecko-reviewers: mstange

rniwa commented on custom-elements/perform-microtask-checkpoint-before-construction-expected.txt:
```diff
@@ -0,0 +1,2 @@
+PASS HTML parser must perform a microtask checkpoint before constructing a custom element
+PASS HTML parser must perform a microtask checkpoint before constructing a custom element during the adoption agency algorithm
```
We shouldn't be adding `-expected` files to WPT. These are WebKit specific files meant to keep track of the current test result.

darsh7807 replied:
Thanks, I have removed it now!

rniwa commented:
This is an importation of a test from https://trac.webkit.org/changeset/234944. In the future, please mention that in the commit even if you kept the author information in the file itself; a proper attribution is essential.

xiaoyu3x committed:
Add CEReactions tests for HTMLButtonElement (#10137)

* Add CEReactions tests for HTMLButtonElement

- Verify attributes of HTMLButtonElement interface have CEReactions
- Adjust functions "testReflectAttribute" and "testReflectAttributeWithContentValues"
in the reactions.js to compatible customized built-in element

mikeconley committed:
Split out prefetch test from fetch-destination.https.html, and disable for Firefox.

Differential Revision: https://phabricator.services.mozilla.com/D12944

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1503852
gecko-commit: d534baf412772ee14e0d89f651267cb4dc2a3f6c
gecko-integration-branch: autoland
gecko-reviewers: jgraham

foolip committed:
Update xhr/abort-during-upload.htm

Remove `document.title` default argument from `async_test` as `document.title` is effectively already used
in testharness.js

Co-Authored-By: qiuzhong <zhongx.qiu@intel.com>

irori committed:
Signed Exchange: Disallow HEAD request method

As per the Loading Signed Exchanges spec [1], this patch makes
SignedExchangeEnvelope::Parse() fail if the exchange's request method is
not "GET".

[1] https://wicg.github.io/webpackage/loading.html#parse-cbor-headers

Bug: 803774
Change-Id: I4729403f3dae5038bae702b0359e1b98f9a11233
Reviewed-on: https://chromium-review.googlesource.com/c/1350017
Reviewed-by: Kouhei Ueno <kouhei@chromium.org>
Reviewed-by: Tsuyoshi Horo <horo@chromium.org>
Commit-Queue: Kunihiko Sakamoto <ksakamoto@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610767}

irori committed:
Rename srcset/imgsizes link attributes to imagesrcset/imagesizes

This patch renames "srcset" and "imgsizes" attributes on link
to "imagesrcset" and "imagesizes" respectively, to match the latest
spec PR [1].

Note: this feature is behind experimental-web-platform-features flag.

[1] https://github.com/whatwg/html/pull/4048

Bug: 813452
Change-Id: I8eabfbd734e2f29e36b7f7a3c4a32f1237b042b1
Reviewed-on: https://chromium-review.googlesource.com/c/1350414
Reviewed-by: Kouhei Ueno <kouhei@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Commit-Queue: Kunihiko Sakamoto <ksakamoto@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610782}

Commit:
Fix CSSPreloadScanner to avoid missing rules (reland)

This fixes a couple of bugs with the CSSPReloadScanner that made it miss
rules in which the URL wasn't quoted or the rule didn't end with a
semicolon. It also adds tentative WPT tests for that functionality, as
the tests are also relevant for WebKit.

This is a reland of
https://chromium-review.googlesource.com/c/chromium/src/+/1331042

Bug: 903785
Change-Id: I401c252a42fbb96dee9c7942e0a4f8d5b6850244
TBR: kouhei
Reviewed-on: https://chromium-review.googlesource.com/c/1349980
Reviewed-by: Yoav Weiss <yoav@yoav.ws>
Commit-Queue: Yoav Weiss <yoav@yoav.ws>
Cr-Commit-Position: refs/heads/master@{#610796}

mrego committed:
[css-contain] Fix baseline for elements with contain: layout

Layout containment elements are treated as having no baseline,
when that's applied to table cells, flex or grid items
that are baseline aligned that means that they should use
their synthesized baseline.

The patch adds simple check in LayoutBlockFlow::FirstLineBoxBaseline()
similar to what we have in other methods. If the element has
layout containment it has no baseline (it returns -1).
The same thing is done in LayoutNGMixin<Base>::FragmentBaseline()
for LayoutNG.

Two tests from the WPT suite has been modified in order to avoid
baseline alignment as they are not checking that feature
of layout containment but something else.

BUG=907709
TEST=external/wpt/css/css-contain/contain-layout-baseline-002.html
TEST=external/wpt/css/css-contain/contain-layout-baseline-003.html
TEST=external/wpt/css/css-contain/contain-layout-baseline-004.html

Change-Id: Id393787dd24bf3e34aaf73476e446fb9d0a4e593
Reviewed-on: https://chromium-review.googlesource.com/c/1348892
Commit-Queue: Manuel Rego <rego@igalia.com>
Reviewed-by: Morten Stenshorne <mstensho@chromium.org>
Reviewed-by: Koji Ishii <kojii@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610833}

andypaicu committed:
Inherit the navigation initiator when navigating instead of the parent/opener

Spec PR: https://github.com/w3c/webappsec-csp/pull/358

Bug: 905301, 894228, 836148
Change-Id: I43ada2266d42d1cd56dbe3c6dd89d115e878a83a
Reviewed-on: https://chromium-review.googlesource.com/c/1314633
Commit-Queue: Andy Paicu <andypaicu@chromium.org>
Reviewed-by: Mike West <mkwst@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610850}

ehsan-karamad committed:
Fix some minor feature policy parsing issues

This CL makes the following changes to the feature policy parsing code:

  1- ParsedFeaturePolicyDeclaration holds a sorted vector of unique
     |origins|.
  2- AllowList uses std::set instead of std::vector.
  3- When parsing for list of origins, in case of matching all origins
     (*), the current set of origins is cleared.
  4- When comparing ParsedFeaturePolicyDeclaration, if both
     declarations include '*' then the set of origins are not compared.

The noticeable outcome of the CL is that parsed policy will ignore
repeated origins and will be sorted. This would make the feature lookup
algorithm more efficient.

Bug: 710324
Change-Id: I5c67ee2d6cff891304781bea0998e07739006a2e
Reviewed-on: https://chromium-review.googlesource.com/c/1161753
Reviewed-by: Jeremy Roman <jbroman@chromium.org>
Reviewed-by: Alex Moshchuk <alexmos@chromium.org>
Reviewed-by: Ian Clelland <iclelland@chromium.org>
Commit-Queue: Ehsan Karamad <ekaramad@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610887}

LanWei22 committed:
Modify the web platform tests with mouse inputs to use TestDriver

Since we have implemented test_driver Actions API, we will replace
chrome.gpuBenchmarking.pointerActionSequence with this Actions API in
all the tests. This CL will just replace the web platform tests with
mouse inputs.

Bug: 893480
Change-Id: Ie4eb7f4b8aaf23b2664182c5b8e814c7113dcf9a
Reviewed-on: https://chromium-review.googlesource.com/c/1296823
Commit-Queue: Lan Wei <lanwei@chromium.org>
Reviewed-by: Navid Zolghadr <nzolghadr@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610898}

irori committed:
Add imageSrcset and imageSizes IDL attributes to HTMLLinkElement

This feature is behind the Experimental Web Platform feature flag.

Spec PR: https://github.com/whatwg/html/pull/4048

Bug: 813452
Change-Id: I84c58035789b64a311637cebc4672d16661980c2
Reviewed-on: https://chromium-review.googlesource.com/c/1350558
Reviewed-by: Dominic Farolino <domfarolino@gmail.com>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Commit-Queue: Kunihiko Sakamoto <ksakamoto@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610990}

mounirlamouri committed:
Revert "Inherit the navigation initiator when navigating instead of the parent/opener"

This reverts commit 108147dfd1ea159fd3632ef92ccc4ab8952980c7.

Reason for revert:
FindIt suggests a 74% chance that this is the cause of the following issues:
https://ci.chromium.org/buildbot/chromium.webkit/WebKit%20Linux%20Trusty%20Leak/26833

Original change's description:
> Inherit the navigation initiator when navigating instead of the parent/opener
>
> Spec PR: https://github.com/w3c/webappsec-csp/pull/358
>
> Bug: 905301, 894228, 836148
> Change-Id: I43ada2266d42d1cd56dbe3c6dd89d115e878a83a
> Reviewed-on: https://chromium-review.googlesource.com/c/1314633
> Commit-Queue: Andy Paicu <andypaicu@chromium.org>
> Reviewed-by: Mike West <mkwst@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#610850}

TBR=mkwst@chromium.org,andypaicu@chromium.org

Change-Id: If3ccf72cf8a4285926429f2855f32c1b0c606c5b
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Bug: 905301, 894228, 836148
Reviewed-on: https://chromium-review.googlesource.com/c/1351285
Reviewed-by: Mounir Lamouri <mlamouri@chromium.org>
Commit-Queue: Mounir Lamouri <mlamouri@chromium.org>
Cr-Commit-Position: refs/heads/master@{#610930}

autofoolip committed:
Update interfaces/uievents.idl

Source: https://github.com/tidoust/reffy-reports/blob/6f56f1e/whatwg/idl/uievents.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/460112731

fred-wang commented:
> This is an importation of a test from https://trac.webkit.org/changeset/234944. In the future, please mention that in the commit even if you kept the author information in the file itself. Because I contribute to WebKit on behalf of Apple, the author of the file and therefore the entity who hols the copyright to the test is Apple. Without a proper hyperlink in the commit message, discovering that attribution is hard unless you know who I am.
> 
> Most of WPT and browser vendor folks probably know or recognize my name one way or another, and can probably infer things from my `@webkit.org` email address but that won't hold true in 20-30 years from now.

Right, I edited the issue title and description to match the behavior of Youenn's export script. We should ensure we have this reference when we squash/merge this.

annevk committed:
Fetch: Content-Type parsing

See https://github.com/whatwg/fetch/pull/831 for context.

foolip committed:
Rewrite `async_test(document.title)` to just `async_test()` (#14232)

If no test name is given, `document.title` is effectively already used
in testharness.js:
https://github.com/web-platform-tests/wpt/blob/0ddb31913184805fa3725e22e5b88046e961996b/resources/testharness.js#L3218-L3224

qiuzhong committed:
Remove timeout from promise_test (#14253)

Related: #11120

autofoolip committed:
Update interfaces/reporting.idl (#14250)

Source: https://github.com/tidoust/reffy-reports/blob/6f56f1e/whatwg/idl/reporting.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/460112731

autofoolip committed:
Update interfaces/scroll-animations.idl (#14251)

Source: https://github.com/tidoust/reffy-reports/blob/6f56f1e/whatwg/idl/scroll-animations.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/460112731

emilio committed:
[css-text] Add a reftest for crbug.com/906369 / webkit.org/b/191807.

emilio committed:
Merge pull request #14258 from emilio/crbug-906369

[css-text] Add a reftest for crbug.com/906369 / webkit.org/b/191807.

ricea committed:
ReadableStream: modify tests for pipeThrough() having a brand check (#14193)

Remove tests that ReadableStream.prototype.pipeThrough operates
generically on its arguments. Add tests that it performs brand checks on
|this|, |readable| and |writable|, and throws for other precondition
failures.

Also verify that it *doesn't* call pipeTo().

Also verify that preventClose, preventCancel and preventAbort work,
since we can no longer test that it passes through to pipeTo().

Also change the expected length of the function from 2 to 1 to reflect
that the options argument is optional.

Corresponding standard changes are at
https://github.com/whatwg/streams/pull/966.

Commit:
Align resource timing buffer full processing to spec PR 168 (reland)

This change implements the processing model from PR 168[1], when
it comes to setResourceTimingBufferSize(), clearResourceTimings()
and the firing of the resourcetimingbufferfull event.

This is a reland of
https://chromium-review.googlesource.com/c/chromium/src/+/1345269
(but with nicer tests).

[1] https://github.com/w3c/resource-timing/pull/168

Change-Id: I10431bfda7f79b484c7ee5c608bb8a360d905339
Bug: 908181, 908414
Reviewed-on: https://chromium-review.googlesource.com/c/1350950
Reviewed-by: Nicolás Peña Moreno <npm@chromium.org>
Commit-Queue: Yoav Weiss <yoavweiss@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611174}

KWierso committed:
Update Fennec wpt expectations

Differential Revision: https://phabricator.services.mozilla.com/D13012

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1499003
gecko-commit: 268c6832d2bebe890b737c14755b3785366cf829
gecko-integration-branch: central
gecko-reviewers: jgraham

dbaron committed:
Sync Mozilla CSS tests as of https://hg.mozilla.org/mozilla-central/rev/ce39a152428a7f8ba5a4c82455dcf501c76c031b . (#14264)

This contains changes from [bug 1507663](https://bugzilla.mozilla.org/show_bug.cgi?id=1507663) by @dholbert, reviewed by @aethanyc.

nhiroki committed:
Worker: Add referrer policy tests for shared workers

I manually changed only following files:

 - external/wpt/referrer-policy/generic/common.js
 - external/wpt/referrer-policy/generic/referrer-policy-test-case.js
 - external/wpt/referrer-policy/generic/subresource/shared-worker.py
 - external/wpt/referrer-policy/generic/template/shared-worker.js.template
 - external/wpt/referrer-policy/spec.src.json
 - TestExpectations

Others were auto-generated.

Bug: 835717, 906959
Change-Id: Ib0a84710fe94ba0893d9609bb708a3f5e8fd4091
Reviewed-on: https://chromium-review.googlesource.com/c/1343582
Commit-Queue: Hiroki Nakagawa <nhiroki@chromium.org>
Reviewed-by: Hiroshige Hayashizaki <hiroshige@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611192}

shampson committed:
Adding tests for RTCQuicStream.

Bug: 874296
Change-Id: I33e79a0641e38b420a23d0ff42933cd26bc0b368
Reviewed-on: https://chromium-review.googlesource.com/c/1341153
Commit-Queue: Seth Hampson <shampson@chromium.org>
Reviewed-by: Steve Anton <steveanton@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611242}

ewilligers committed:
[css-multicol] property parsing, computed values (#14259)

Test that CSS Multi-column Layout properties parse exactly the grammar
specified in the spec.
https://drafts.csswg.org/css-multicol/

Test that computed values are as specified, with lengths made absolute.

tabatkins committed:
Merge pull request #14184 from ewilligers/defined-selector

[selectors] :defined CSS selector

ericwilligers committed:
CSS: WPTs for CSS Fragmentation parsing

Test parsing, serialiation and computed values for properties defined
in CSS Fragmentation Module Level 3
https://drafts.csswg.org/css-break/

Blink has expected failures for keywords relating to CSS Regions.

Change-Id: Ie5a930ba2d805c7bd45aa84c293bc8ff5922da98
Reviewed-on: https://chromium-review.googlesource.com/c/1351190
Commit-Queue: Emil A Eklund <eae@chromium.org>
Reviewed-by: Emil A Eklund <eae@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611257}

fippo committed:
webrtc-wpt: use addTrack(track, stream) to increase firefox compat

Firefox does not support addTrack(track). Add a track whereever it makes sense.
The addTrack tests itself might rightfully use this.

The following grep shows most affected places:
  git grep addTrack *.html | grep -v , | grep -v "\.\.\."

Bug: None
Change-Id: Ib225e6d51184c3ccc446ccf93447e2ac7be080c3
Reviewed-on: https://chromium-review.googlesource.com/c/1346394
Reviewed-by: Henrik Boström <hbos@chromium.org>
Commit-Queue: Philipp Hancke <philipp.hancke@googlemail.com>
Cr-Commit-Position: refs/heads/master@{#611262}

ericwilligers committed:
CSS: Serialize contain keywords in canonical order

Keywords size, layout, style, paint now serialize in that order
for the 'contain' property.
https://drafts.csswg.org/css-contain/#contain-property

Bug: 908353
Change-Id: Ifde635f2d313add887dd43a0a0d98249e45bff4b
Reviewed-on: https://chromium-review.googlesource.com/c/1350562
Commit-Queue: Eric Willigers <ericwilligers@chromium.org>
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611270}

ericwilligers committed:
CSS: WPTs for Positioned Layout parsing

Test parsing and computed value for
https://drafts.csswg.org/css-position/

- bottom left right top
- position
- z-index
- inset-after inset-before inset-end inset-start (not currently implemented)

Change-Id: I586271fe584fadb19e5683a2781c32142fd9001a
Reviewed-on: https://chromium-review.googlesource.com/c/1350409
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Commit-Queue: Eric Willigers <ericwilligers@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611349}

ericwilligers committed:
SVG: clamp *-opacity computed value to [0,1]

https://svgwg.org/svg2-draft/painting.html#FillOpacity
https://svgwg.org/svg2-draft/painting.html#StrokeOpacity
https://drafts.fxtf.org/filter-effects/#FloodOpacityProperty
Computed value:	the specified value converted to a number, clamped to the range [0,1]

BUG=908058

Change-Id: I2cb10017c1a42eb991af9ae31c03895847d7f52b
Reviewed-on: https://chromium-review.googlesource.com/c/1350369
Commit-Queue: Eric Willigers <ericwilligers@chromium.org>
Reviewed-by: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611431}

Commit:
Quota: Send usage details in return value of estimate().

Change-Id: I85fd510b4245f966e32340b9d433f7d9b96b3810
Bug: 904000
Reviewed-on: https://chromium-review.googlesource.com/c/1303041
Commit-Queue: Jarryd Goodman <jarrydg@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Reviewed-by: Kent Tamura <tkent@chromium.org>
Reviewed-by: Victor Costan <pwnall@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611491}

Commit:
Only create an ib-split sequence for truly block-outside children.

This changes the logic from !IsInlineOutside() to IsBlockOutside()
so that table-internal display types etc are now excluded
(i.e. no longer creates an ib-split).

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1506895
gecko-commit: 639b931122292b83f5ee99b646942c7352bef2fc
gecko-integration-branch: mozilla-inbound
gecko-reviewers: emilio

fergald committed:
Tidy layout tests for css-shadow-parts.

For consistency with https://crrev.com/c/1333014
- remove type="text/javascript"
- replace "var" with "const"

Change-Id: Ie809fa22763494fd6ca4b20112d4a3e6afdd1f0b
Reviewed-on: https://chromium-review.googlesource.com/c/1350431
Reviewed-by: Hayato Ito <hayato@chromium.org>
Commit-Queue: Fergal Daly <fergal@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611511}

marcoscaceres committed:
Test PaymentMethodData data member is converted in constructor (#14205)

fergald committed:
[css] Switch part attribute to a DOMTokenList

Adds tests for manipulating it via the attribute.

Renames PartNames() to Part() to match with part(), the API wrapper.

This uses the same IDL as classList.

Bug: 805271
Change-Id: I9715df48478510029f5d985124e46111c41cffc3
Reviewed-on: https://chromium-review.googlesource.com/c/1333014
Commit-Queue: Fergal Daly <fergal@chromium.org>
Reviewed-by: Kent Tamura <tkent@chromium.org>
Reviewed-by: Hayato Ito <hayato@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611570}

moz-wptsync-bot committed:
Part 4 - mochitest for PaymentMethodChangeEvent. (#14272)

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1477117
gecko-commit: 3b473ca68a5777d754b2a9f79275497309eee9fb
gecko-integration-branch: central
gecko-reviewers: baku

andypaicu committed:
Added srcdoc iframe inheritance test

Added a test for behavior around srcdoc iframes CSP inheritance.
This tests the current specified behavior of always inheriting the
parent CSP for srcdoc iframes.

Bug: 694525
Change-Id: I049ef8c5a9e75c052dc2767ea2d523f54cca497f
Reviewed-on: https://chromium-review.googlesource.com/c/1350889
Reviewed-by: Mike West <mkwst@chromium.org>
Commit-Queue: Andy Paicu <andypaicu@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611638}

andypaicu committed:
Add javascript navigations interop tests

https://github.com/w3c/webappsec-csp/issues/322 raises an interop issue
It appears the issue itself has been fixed since but I've added these
tests regardless to ensure there is no regression.

Bug: 694525
Change-Id: Icb5502e228b3a96c176cbe23d5b4ce9c5c8640ab
Reviewed-on: https://chromium-review.googlesource.com/c/1348054
Reviewed-by: Andy Paicu <andypaicu@chromium.org>
Reviewed-by: Mike West <mkwst@chromium.org>
Commit-Queue: Andy Paicu <andypaicu@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611643}

alijuma committed:
WebKit export of https://bugs.webkit.org/show_bug.cgi?id=191975 (#14266)

* WebKit export of https://bugs.webkit.org/show_bug.cgi?id=191975
* Intersection Observer: rootMargin: '' gives weird results

nedn committed:
Revert "Align resource timing buffer full processing to spec PR 168 (reland)"

This reverts commit 6dd060768d1c1183e0d29e422a2213f106d98299.

Reason for revert: suspect causing  WebRtcDisableEncryptionFlagBrowserTest.VerifyEncryption to be flaky (FindIt's report)

BUG:chromium:909704

Original change's description:
> Align resource timing buffer full processing to spec PR 168 (reland)
>
> This change implements the processing model from PR 168[1], when
> it comes to setResourceTimingBufferSize(), clearResourceTimings()
> and the firing of the resourcetimingbufferfull event.
>
> This is a reland of
> https://chromium-review.googlesource.com/c/chromium/src/+/1345269
> (but with nicer tests).
>
> [1] https://github.com/w3c/resource-timing/pull/168
>
> Change-Id: I10431bfda7f79b484c7ee5c608bb8a360d905339
> Bug: 908181, 908414
> Reviewed-on: https://chromium-review.googlesource.com/c/1350950
> Reviewed-by: Nicolás Peña Moreno <npm@chromium.org>
> Commit-Queue: Yoav Weiss <yoavweiss@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#611174}

TBR=npm@chromium.org,yoavweiss@chromium.org

Change-Id: Ia4bf26c6be1e7c4ae35b904bac25fe031f854604
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Bug: 908181, 908414
Reviewed-on: https://chromium-review.googlesource.com/c/1354123
Reviewed-by: Ned Nguyen <nednguyen@google.com>
Commit-Queue: Ned Nguyen <nednguyen@google.com>
Cr-Commit-Position: refs/heads/master@{#611722}

otherdaniel committed:
[Trusted Types] Prevent mutation of Trusted Types built-ins.

Bug: 739170
Change-Id: Iafeabd00f6dde0c1ecf055a5f9f006059e43c9f7
Reviewed-on: https://chromium-review.googlesource.com/c/1346449
Commit-Queue: Daniel Vogelheim <vogelheim@chromium.org>
Reviewed-by: Mike West <mkwst@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611736}

JohnChen0 committed:
Remove browserName from default Chrome capabilities

The configuration for Chrome currently sends 'browserName' capability
to ChromeDriver by default. This is unnecessary, and causes errors in
several test cases in webdriver/tests/new_session, whenever a test case
wants to set a different browserName.

autofoolip committed:
Update interfaces/web-nfc.idl (#14277)

Source: https://github.com/tidoust/reffy-reports/blob/6f59720/whatwg/idl/web-nfc.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/460633569

autofoolip committed:
Update interfaces/css-fonts.idl (#14275)

Source: https://github.com/tidoust/reffy-reports/blob/6f59720/whatwg/idl/css-fonts.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/460633569

Commit:
Disabled (empty) SVG shapes cannot be hit

Disabled SVG shapes [1][2][3] should not be possible to hit since they
are considered "not rendered".

[1] https://svgwg.org/svg2-draft/shapes.html#RectElement
[2] https://svgwg.org/svg2-draft/shapes.html#CircleElement
[3] https://svgwg.org/svg2-draft/shapes.html#EllipseElement

Bug: 896641
Change-Id: I5fe530722a4d60ecd416cd68f06a7555bb6a520f
Reviewed-on: https://chromium-review.googlesource.com/c/1353983
Reviewed-by: Philip Rogers <pdr@chromium.org>
Commit-Queue: Fredrik Söderquist <fs@opera.com>
Cr-Commit-Position: refs/heads/master@{#611840}

NavidZ committed:
Add actions builder API documentation

Add actions builder API documentation plus some examples

mounirlamouri committed:
Media Capabilities: implement Blink shell of encrypted media support.

This is adding the interface changes, the code checking for corectedness,
does type conversion and checks for the edge cases mentioned in spec.

This is also adding WPT.

Bug: 907909
Change-Id: Ieebe5b25202a403e50a09576ac283690c0f0ba55
Reviewed-on: https://chromium-review.googlesource.com/c/1348972
Commit-Queue: Mounir Lamouri <mlamouri@chromium.org>
Reviewed-by: Chrome Cunningham <chcunningham@chromium.org>
Reviewed-by: Philip Jägenstedt <foolip@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611871}

Hexcles committed:
Attempt to monitor memory from userspace

It is suspected that Taskcluster jobs sometimes fail due to OOM. It is
hard to know what exactly happens if the kernel OOM killer kicks in.
This change attempts to use a user-space solution: earlyoom.

There are two benefits of earlyoom:
1. Logging: we can log the memory usage periodically and the actions
   taken by earlyoom when the system is low on memory.
2. Recovery: we can teach earlyoom to prefer to kill browser processes
   and avoid python processes (wpt) so that the runner has a chance to
   recover (marking the test as CRASH and carrying on).

mounirlamouri committed:
Media Capabilities: implement response side of Encrypted Media.

This is adding an DecodingInfo class that carries key system access
information back and create a MediaKeySystemAccess.

This implementation in //media/blink is a stub that is meant to only
pass WPT tests at the moment.

Bug: 907909
Change-Id: I3bac0087ca3051defb8b47d10fffdb6052814ae9
Reviewed-on: https://chromium-review.googlesource.com/c/1348733
Commit-Queue: Mounir Lamouri <mlamouri@chromium.org>
Reviewed-by: Chrome Cunningham <chcunningham@chromium.org>
Cr-Commit-Position: refs/heads/master@{#611917}

rniwa committed:
Move back resources/ustom-elements-helpers.js into custom-elements/resources/custom-elements-helpers.js after darsh7807 (#14295)

This file has no business being in the top-level resources directory.

autofoolip committed:
Update interfaces/webxr.idl (#14013)

Source: https://github.com/tidoust/reffy-reports/blob/a232594/whatwg/idl/webxr.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/461143420

autofoolip committed:
Update interfaces/payment-request.idl (#14296)

Source: https://github.com/tidoust/reffy-reports/blob/a232594/whatwg/idl/payment-request.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/461143420

svgeesus committed:
Merge pull request #14111 from ewilligers/svg-text-parsing

SVG Text Parsing

npm1 committed:
Move EventTiming tests to WPT Try 3

This CL moves the tests in http/tests/event-timing to external/wpt/event-timing.
The slow image is change from php based to python based. The click is now
handled by test driver. and setTimeout is replaced with step_timeout.

Bug: 841224, 908187, 907948, 907949
Change-Id: Ia6359878666cfb96645aa3b7cd7736ce9df97683
Reviewed-on: https://chromium-review.googlesource.com/c/1352617
Reviewed-by: Timothy Dresser <tdresser@chromium.org>
Commit-Queue: Nicolás Peña Moreno <npm@chromium.org>
Cr-Commit-Position: refs/heads/master@{#612214}

emilio committed:
Prevent exponential blowup of custom properties.

Put a hard cap on the value length instead of counting substitutions, because it
works best, see the comment.

Differential Revision: https://phabricator.services.mozilla.com/D13352

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1510862
gecko-commit: e593d06bcd929ab543824921434def8c6e4c6c5b
gecko-integration-branch: central
gecko-reviewers: jwatt

NavidZ committed:
Remove trailing space and add a "the"

darsh7807 committed:
Corrected the file path of custon-elements-helper.js

darsh7807 committed:
Corrected the file path of custon-elements-helper.js

darsh7807 committed:
Removed expected txt file

darsh7807 committed:
Removed expected txt file

darsh7807 committed:
Corrcted formatting

darsh7807 committed:
Corrcted formatting

jugglinmike committed:
[permissions] Add tests for "Set Permission" (#12118)

https://w3c.github.io/permissions/#automation

mfreed7 committed:
Adding more WPT tests for backdrop-filter

This set includes a few more known issues, and changes the definition of
opacity to match our more recent proposal. Under that proposal, the
opacity of the backdrop-filter element applies to the filtered backdrop
image as well.

Bug: 497522
Change-Id: I2fa0bbb710b7a4ba5592c9a4b9ea5ecaaf5b9ddf
Reviewed-on: https://chromium-review.googlesource.com/c/1354347
Reviewed-by: Chris Harrelson <chrishtr@chromium.org>
Commit-Queue: Mason Freed <masonfreed@chromium.org>
Cr-Commit-Position: refs/heads/master@{#612325}

youennf committed:
WebKit export of https://bugs.webkit.org/show_bug.cgi?id=191986 (#14286)

rniwa commented:
I've already moved `custom-elements-helpers.js` in https://github.com/web-platform-tests/wpt/pull/14295 so you'd have to revert that change.

qiuzhong committed:
Remove timeout from async_test for argument form. (#14273)

There're cases that the timeout is passing to async_test in the tests.
They should be removed as well.
Related: #11120

acomminos committed:
Add web platform test for verifying first byte timing of responseStart

Verfies that the UA measures the time before the end of header parsing
in responseStart by simulating a delay after writing the status line and
flushing to the client. Chrome fails this test for now, but Firefox
passes.

Bug: 568024
Change-Id: I61569c6a5b0d8ea2f3a06cb37a3d1ed36c06d40f
Reviewed-on: https://chromium-review.googlesource.com/c/1354645
Commit-Queue: Andrew Comminos <acomminos@fb.com>
Reviewed-by: Nicolás Peña Moreno <npm@chromium.org>
Reviewed-by: Yoav Weiss <yoavweiss@chromium.org>
Cr-Commit-Position: refs/heads/master@{#612412}

hwanseung committed:
[GeometryInterface] add scaleNonUniform method in DOMMatrixReadOnly

DOMMatrix should be alias to SVGMatrix. (It was not implement in blink yet)
SVGMatrix have scaleNonUniform() function. but DOMMatrix was not.
there are some website which is used SVGMatrix.scaleNonUniform().
it would be possible brake some website when DOMMatrix be alias
to SVGMatrix.

Intent to implement and ship:
https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/BJJKDby5p7w

Bug: 905921
Change-Id: Ic09b8d6e46d0aa822b766bd6cfbc5ce7c2164849
Reviewed-on: https://chromium-review.googlesource.com/c/1341279
Commit-Queue: Hwanseung Lee <hs1217.lee@samsung.com>
Reviewed-by: Philip Jägenstedt <foolip@chromium.org>
Reviewed-by: Fernando Serboncini <fserb@chromium.org>
Reviewed-by: Jinho Bang <jinho.bang@samsung.com>
Cr-Commit-Position: refs/heads/master@{#612473}

andreas-butler committed:
[IndexedDB]: Implement explicit commit() in renderer

Implementing an explicit commit function for the IndexedDB Transaction API.
The addition of this API allows developers to preempt IndexedDB's autocommit
functionality by sending their own explicit commit signal.

Explainer: https://andreas-butler.github.io/idb-transaction-commit/EXPLAINER
Bug: https://bugs.chromium.org/p/chromium/issues/detail?id=898257
Change-Id: I5e8bdc697052139d014757408d9a7f2b6367655b
Reviewed-on: https://chromium-review.googlesource.com/c/1300055
Commit-Queue: Andreas Butler <andreasbutler@google.com>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Reviewed-by: Victor Costan <pwnall@chromium.org>
Reviewed-by: Daniel Murphy <dmurph@chromium.org>
Cr-Commit-Position: refs/heads/master@{#612474}

jyasskin committed:
Test navigation when a SW replies with a fetch from a different URL (#14288)

The resulting document's location should be the originally requested
URL. See also https://github.com/whatwg/html/pull/4205

janvarga committed:
Part 14: Enhance clearStoragesForPrincipal() to support clearing of storages for specific quota client

See also bug 1402254, original patch by baku.

bugzilla-url: https://bugzilla.mozilla.org/show_bug.cgi?id=1286798
gecko-commit: 20606f9de28fd7dff30f30b3862b2bdc68df85b3
gecko-integration-branch: central
gecko-reviewers: asuth

svgeesus committed:
Merge pull request #14310 from web-platform-tests/convert_svg11

Convert SVG 1.1 marker tests to reftests.

autofoolip committed:
Update interfaces/screen-capture.idl (#14308)

Source: https://github.com/tidoust/reffy-reports/blob/1a7aa08/whatwg/idl/screen-capture.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/461650522

darsh7807 committed:
Changed the file as in upstream

darsh7807 committed:
Resolved merge conflicts

darsh7807 commented:
Created new pull request, https://github.com/web-platform-tests/wpt/pull/14314 
I messed up this one!
